hotel-management/src/pages/FormPage/index.tsx

```diff
@@ -1,37 +1,53 @@
 import { useMatch, useParams } from 'react-router-dom';
 
 import { RoomForm } from '@/components/RoomForm';
+import { Spinner } from '@/components/Spinner';
+import { useFetchRoom } from '@/hooks/room';
 import { FORM_ACTION, ROUTES } from '@/constants';
 import { Room } from '@/types';
 
 import './index.css';
 
 export const FormPage = () => {
+  const params = useParams<{ id?: string }>();
+  const { id } = params;
   const { ADD, EDIT } = FORM_ACTION;
   const editMatch = useMatch(ROUTES.EDIT_ROOM_PAGE);
-  const params = useParams<{ id?: string }>();
-  const formAction = params.id && editMatch ? EDIT : ADD;
-
-  let room: Room = {
-    id: '',
-    name: '',
-    price: 0,
-    type: 'single',
-    isAvailable: true,
-    description: '',
-    imgUrl: '',
-    amenities: [],
+  const formAction = id && editMatch ? EDIT : ADD;
+  const { isLoading, room } =
+    formAction === EDIT && id
+      ? useFetchRoom(id)
+      : { isLoading: false, room: null };
```

Comment:
This is not good, we can define a flag `enable` inside the hook call, if `enable` is `false` the api is not call and return default value